cli/src/main.rs

```diff
@@ -38,7 +38,15 @@ fn main() {
 
     setup_panic_hooks();
 
-    let mut session = Session::init().unwrap();
+    let session_result = Session::init();
+
+    if let Err(err) = session_result {
+        report(&err);
+        let code = err.exit_code();
+        code.exit();
```

Comment:
Though this will exit the program, should you still put a `return code.exit();` here to signal that this is an early termination of the progression?

Reply:
@jbaxleyiii if you throw the `return` in front of that, I get warnings in the build about `unreachable` code. However, if you _remove_ the semi-colon, you do not get those warnings...

Reply:
So I just tried doing this and our the standard Rust Formatter for the file keeps changing it to `return code.exit();`...and then the warnings start popping up. Not saying this to discourage doing this (I committed things and pushed) but simply want a call out that we might need to define our rust fmt config to prevent folks' tools from overwriting this etc.